source/extensions/filters/network/mysql_proxy/mysql_decoder.cc

```diff
@@ -39,36 +39,51 @@ void DecoderImpl::parseMessage(Buffer::Instance& message, uint8_t seq, uint32_t
     }
     break;
   }
-
   case MySQLSession::State::SslPt:
     // just consume
     message.drain(len);
     break;
-
   case MySQLSession::State::ChallengeResp41:
   case MySQLSession::State::ChallengeResp320: {
-    ClientLoginResponse client_login_resp{};
-    client_login_resp.decode(message, seq, len);
-    callbacks_.onClientLoginResponse(client_login_resp);
-
-    switch (client_login_resp.getRespCode()) {
+    uint8_t resp_code;
+    if (BufferHelper::peekUint8(message, resp_code) != DecodeStatus::Success) {
+      session_.setState(MySQLSession::State::NotHandled);
+      break;
+    }
+    switch (resp_code) {
```

Comment:
This is great! If you want to simplify it even further you could do:

`
```
std::unique_ptr<ClientLoginResponse> msg;
case MYSQL_RESP_OK:
  msg = make_unique<...> OKMessage();
  state = MySQLSession::State::Req;
  break;
case MYSQL_RESP_AUTH_SWITCH:
  msg = make_unique<...> AuthSwithMessage();
  state = MySQLSession::State::AuthSwitchResp;
  break;

....

// Invoke parser and set state.
msg->decode(message, seq, len);
callbacks_onClientLoginResponse(msg);
session_.setState(state);
```
`
But this is up to you.

Reply:
It simplify the code, fixed.